avogadro/core/graph.cpp

```diff
@@ -363,8 +357,8 @@ void Graph::removeEdges(size_t index)
     m_subgraphDirty[m_vertexToSubgraph[index]] = true;
 
   const std::vector<size_t>& edges = m_edgeMap[index];
-  for (size_t i = 0; i < edges.size(); ++i)
```

Comment:
Here's the bug. If you put this back, the test will pass:
```
  for (size_t i = 0; i < edges.size(); ++i)
    removeEdge(edges[i]);
```

I always get nervous with iterators that remove things. I'd have to think carefully about why the range iterator is failing, but the simple solution is to just revert to the old-school version.